.github/workflows/nvidia-test.yml

```diff
@@ -31,11 +30,25 @@ jobs:
             - name: Checkout code
               uses: actions/checkout@v4
 
+            - name: Get the artifact
+              uses: dawidd6/action-download-artifact@v6
+              with:
+                workflow: build.yml
```

Comment:
Can this be replaced with `run_id: ${{ inputs.run_id }}` to make the relation between workflows less fragile?

Reply:
Actually, how can the action know which build run to look for based on just the workflow file name?

In [this case](https://github.com/canonical/docker-snap/actions/runs/10557474952/job/29245052143), the Nvidia workflow can't find the right artifact:
```
Run dawidd6/action-download-artifact@v6
==> Repository: canonical/docker-snap
==> Artifact name: docker_10555346598.snap
==> Local path: ./
==> Workflow name: build.yml
==> Workflow conclusion: success
==> Allow forks: false
==> (found) Run ID: 105573532[18](https://github.com/canonical/docker-snap/actions/runs/10557474952/job/29245052143#step:4:19)
==> (found) Run date: 2024-08-26T09:33:57Z
==> (not found) Artifact: docker_10555346598.snap
==> Found the following artifacts instead:
	==> (found) Artifact: docker_10557353[21](https://github.com/canonical/docker-snap/actions/runs/10557474952/job/29245052143#step:4:22)8.snap
```

Replacing that with run id should resolve this.

Reply:
Added here: https://github.com/canonical/docker-snap/pull/174/files